src/sentry/static/sentry/app/components/assistant/guideAnchor.jsx

```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import React from 'react';
+import $ from 'jquery';
+import createReactClass from 'create-react-class';
+import Reflux from 'reflux';
+import {registerAnchor, unregisterAnchor} from '../../actionCreators/guides';
+import GuideStore from '../../stores/guideStore';
+
+// A guide anchor provides a ripple-effect on an element on the page to draw attention
+// to it. Guide anchors register with the guide store, which uses this information to
+// determine which guides can be shown on the page.
+const GuideAnchor = createReactClass({
+  propTypes: {
+    target: PropTypes.string.isRequired,
+    type: PropTypes.string.isRequired,
```

Comment:
Is the type always going to be "text" as it is now? If yes, I wouldn't pass it as a prop. If there would be other types I'd suggest `PropTypes.oneOf(['text', 'somethingelse'])` just to be super clear when looking at this component about what values are allowed there

Reply:
Yeah there's a button type as well, will use `oneOf`